optuna/samplers/_tpe/parzen_estimator.py

```diff
@@ -184,20 +184,29 @@ def log_pdf(self, samples_dict: Dict[str, np.ndarray]) -> np.ndarray:
                 assert mus is not None
                 assert sigmas is not None
 
-                cdf_func = _ParzenEstimator._normal_cdf
-                p_accept = cdf_func(high, mus, sigmas) - cdf_func(low, mus, sigmas)
                 if q is None:
-                    distance = samples[:, None] - mus
-                    mahalanobis = distance / np.maximum(sigmas, EPS)
-                    z = np.sqrt(2 * np.pi) * sigmas
-                    coefficient = 1 / z / p_accept
-                    log_pdf = -0.5 * mahalanobis**2 + np.log(coefficient)
+                    log_pdf = stats.truncnorm.logpdf(
+                        samples[:, None],
+                        (low - mus) / sigmas,
+                        (high - mus) / sigmas,
+                        loc=mus,
+                        scale=sigmas,
+                    )
                 else:
+                    # TODO(nzw0301): Simplify the logic by following
+                    # https://github.com/optuna/optuna/pull/3985#issuecomment-1253637444
+                    # when we can assume scipy 1.9.2 is commonly used, which includes
+                    # accurate logcdf https://github.com/scipy/scipy/pull/17064.
+                    cdf_func = _ParzenEstimator._trunc_normal_cdf
+                    p_accept = cdf_func(high, mus, sigmas, low, high) - cdf_func(
+                        low, mus, sigmas, low, high
+                    )
```

Comment:
I expect `p_accept` always is 1. Is it right?

Reply:
Sorry for my late response. I'll revisit the PR in a week.

Reply:
Again, sorry for my late response. I suppose so. Can we replace `p_accept` with `1`? I'm not familiar with TPE's calculation, so I'm a little bit lost...

Reply:
`cdf_func(high, mus, sigmas, low, high)` is equal to `stats.truncnorm.cdf(high, (low-mus)/sigmas, (high-mus)/sigmas, loc=mu, scales=sigmas)`, and it is 1.
Similarly , `cdf_func(low, mus, sigmas, low, high)` is 0.